v-next/hardhat/test/internal/builtin-plugins/network-manager/network-manager.ts

```diff
@@ -355,431 +363,1943 @@ describe("NetworkManagerImplementation", () => {
+                  "Expected a hex-encoded private key or a Configuration Variable",
+              },
+              {
+                path: ["networks", "hardhat", "accounts", 5, "balance"],
+                message: "Expected a string or a positive bigint",
+              },
+              {
+                path: ["networks", "hardhat", "accounts", 5, "privateKey"],
+                message:
+                  "Expected a hex-encoded private key or a Configuration Variable",
+              },
+            ]);
+          });
+
+          it("should fail when the array of objects contains an invalid private key", async () => {
+            const validationErrors = await validateNetworkUserConfig(
+              edrConfig({
+                accounts: [{ privateKey: "0xxxxx", balance: 213n }],
+              }),
+            );
+
+            assertValidationErrors(validationErrors, [
+              {
+                path: ["networks", "hardhat", "accounts", 0, "privateKey"],
+                message:
+                  "Expected a hex-encoded private key or a Configuration Variable",
+              },
+            ]);
           });
 
-          it("should not allow invalid private keys", async () => {
-            networkConfig.accounts = [
-              "0xgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg",
+          it("should fail with invalid HD accounts", async () => {
+            const accountsValuesToTest: Array<
+              [Partial<any>, ExpectedValidationError[]]
+            > = [
+              [
+                { mnemonic: 123 },
+                [
+                  {
+                    path: ["networks", "hardhat", "accounts", "mnemonic"],
+                    message: "Expected a string or a Configuration Variable",
+                  },
+                ],
+              ],
+              [
+                { mnemonic: "valid", initialIndex: "asd" },
+                [
+                  {
+                    path: ["networks", "hardhat", "accounts", "initialIndex"],
+                    message: "Expected number, received string",
+                  },
+                ],
+              ],
+              [
+                { mnemonic: "valid", initialIndex: 1, count: "asd" },
+                [
+                  {
+                    path: ["networks", "hardhat", "accounts", "count"],
+                    message: "Expected number, received string",
+                  },
+                ],
+              ],
+              [
+                { mnemonic: "valid", initialIndex: 1, count: 1, path: 123 },
+                [
+                  {
+                    path: ["networks", "hardhat", "accounts", "path"],
+                    message: "Expected string, received number",
+                  },
+                ],
+              ],
+              // todo: is this expected? the zod type has every field optional, so passing an object with an unrelated field doesn't fail
+              [
```

Comment:
I believe this is Zod’s default behavior for unrecognized keys in an object:

> By default, Zod object schemas strip out unrecognized keys during parsing. You can disallow unknown keys with .strict().

I don’t see this as an issue because extra keys won’t be used by hardhat and might even be needed by plugins in the future. Enabling `.strict()` would prevent plugins from extending the configuration.